### PR TITLE
 Add C implementation for `@stdlib/number/float64/base/signbit`

### PR DESCRIPTION
**Pull request by Pranavchiku**
## Checklist

- [x] include/stdlib/number/float64/base
- [x] include.gypi
- [x] manifest.json
- [x] binding.gyp
- [x] src
- [x] examples
- [x] lib
- [x] benchmark
- [x] test
- [x] update readme.md
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
@kgryte this PR can have a final review :)

File: lib/node_modules/@stdlib/number/float64/base/signbit/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/README.md]:
```suggestion
int8_t stdlib_base_float64_signbit( const double x );
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/README.md]:
```suggestion
Returns an integer indicating whether the sign bit for a double-precision floating-point number is on (`1`) or off (`0`).
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/README.md]: Nit: need to use macro for printing `int8_t` values.

File: lib/node_modules/@stdlib/number/float64/base/signbit/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/README.md]:
```suggestion
<!-- /.c -->
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/benchmark.native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/benchmark.native.js]:
```suggestion
bench( pkg+'::native', opts, function benchmark( b ) {
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/benchmark.native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/benchmark.native.js]:
```suggestion
		if ( y !== 1 && y !== 0 ) {
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/benchmark.native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/benchmark.native.js]:
```suggestion
	if ( y !== 1 && y !== 0 ) {
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/c/benchmark.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/c/benchmark.c]:
```suggestion
		if ( out != 0 && out != 1 ) {
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/examples/c/example.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/examples/c/example.c]: Nit: use macro for `int8_t`

File: lib/node_modules/@stdlib/number/float64/base/signbit/include/stdlib/number/float64/base/signbit.h
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/include/stdlib/number/float64/base/signbit.h]: Update description and signature.

File: lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js:33]:
```suggestion
* @private
* @param {number} x - input value
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js]:
```suggestion
* @returns {boolean} - boolean indicating if sign bit is on or off
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js]:
```suggestion
	return Boolean( addon( x ) );
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js]:
```suggestion
var Boolean = require( '@stdlib/boolean/ctor' );
var addon = require( './../src/addon.node' );
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/src/signbit.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/src/signbit.c]: Update desc and signature.

File: lib/node_modules/@stdlib/number/float64/base/signbit/src/signbit.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/src/signbit.c]:
```suggestion
* @param x       input value
* @return          result
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/src/signbit.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/src/signbit.c:38]:
```suggestion

	// Extract from the input value a higher order word (unsigned 32-bit integer) containing the exponent and sign:
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/test/test.native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/test/test.native.js:33]: To compile native add-ons, use the following command:

```bash
make install-node-addons NODE_ADDONS_PATTERN=float64/base/signbit
```

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this, @Pranavchiku! Just some minor nits. Once resolved, this should be ready for merge!

**Comment by Pranavchiku** [comment]
I guess all the changes are done, and when I try to run the `test.native.js` locally, it throws me the following log
```
(base) pranavchiku@pranavchiku-Yoga-Slim-7-14IIL05:~/stdlib(signbit1)$ node /home/pranavchiku/stdlib/lib/node_modules/@stdlib/number/float64/base/signbit/test/test.native.js
TAP version 13
# main export is a function
ok 1 /home/pranavchiku/stdlib/lib/node_modules/@stdlib/number/float64/base/signbit/test/test.native.js
ok 2 main export is a function
# the function returns a boolean
node: symbol lookup error: /home/pranavchiku/stdlib/lib/node_modules/@stdlib/number/float64/base/signbit/src/addon.node: undefined symbol: napi_create_int8
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/src/addon.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/src/addon.c]:
```suggestion
	status = napi_create_int32( env, out, &v );
```

N-API does not have an `int8` API. `int32` should be fine to use here.

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/base/signbit/src/addon.c]: ohh okay, got it :)

Comment (kgryte) [reply lib/node_modules/@stdlib/number/float64/base/signbit/src/addon.c]: For reference: https://nodejs.org/api/n-api.html#functions-to-convert-from-c-types-to-node-api

File: lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/lib/native.js]:
```suggestion
var Boolean = require( '@stdlib/boolean/ctor' );
```

**Comment by Pranavchiku** [comment]
Okay so, I am currently having my previously generated `addon.node`, and it has`napi_create_int8`, when I run that file again it throws me error, regarding `#include "stdlib/number/float64/base/signbit.h"`, how may I update that `addon.node`?

**Comment by kgryte** [comment]
Just rerun the command to build the addon. It will remove the old one and rebuild.

**Comment by Pranavchiku** [comment]
>Just rerun the command to build the addon. It will remove the old one and rebuild.

It generated during the time I pushed the files to PR, else to run locally I used `gcc <path>` which throws me error.

**Comment by kgryte** [comment]
For running locally, you should also use "make install-node-addons". This is necessary in order to resolve headers and deps.

**Comment by Pranavchiku** [comment]
Wait let me check, I already did that, trying it out again.

**Comment by kgryte** [comment]
In the project, we don't compile with gcc directly. We use GYP, which is what Node.js uses as its meta build tool.

**Comment by Pranavchiku** [comment]
Yeah, all Test cases passed, thanks @kgryte!

**Comment by Pranavchiku** [comment]
This PR can have a final review :)

File: lib/node_modules/@stdlib/number/float64/base/signbit/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/README.md]:
```suggestion
#### stdlib_base_float64_signbit( x )
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/README.md]:
```suggestion
stdlib_base_float64_signbit( 3.14 );
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/README.md]:
```suggestion
#include "stdlib/number/float64/base/signbit.h"
#include <stdint.h>
#include <stdio.h>
#include <inttypes.h>

int main() {
    double x[] = { 3.14, -3.14, 0.0, -0.0, 4.0, 1.0, -1.0, 1.0e308, -1.0e308 };

    int8_t out;
    int i;
    for ( i = 0; i < 9; i++ ) {
        stdlib_base_float64_signbit( x[ i ], &out );
        printf( "%lf => signbit: %" PRId8 "\n", x[ i ], out );
    }
}
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/c/benchmark.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/benchmark/c/benchmark.c]:
```suggestion
	if ( out != 0 && out !== 1 ) {
```

C99 [`signbit`](https://en.cppreference.com/w/c/numeric/math/signbit) is not guaranteed to return `1` for negative values.

File: lib/node_modules/@stdlib/number/float64/base/signbit/examples/c/example.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/examples/c/example.c]:
```suggestion
#include "stdlib/number/float64/base/signbit.h"
#include <stdint.h>
#include <stdio.h>
#include <inttypes.h>

int main() {
	double x[] = { 3.14, -3.14, 0.0, -0.0, 4.0, 1.0, -1.0, 1.0e308, -1.0e308 };

	int8_t out;
	int i;
	for ( i = 0; i < 9; i++ ) {
		out = stdlib_base_float64_signbit( x[ i ] );
		printf( "%lf => signbit: %" PRId8 "\n", x[ i ], out );
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/src/addon.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/src/addon.c]:
```suggestion
	int8_t out = stdlib_base_float64_signbit( value );
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/src/addon.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/src/addon.c]:
```suggestion
	status = napi_create_int32( env, (int32_t)out, &v );
```

File: lib/node_modules/@stdlib/number/float64/base/signbit/src/signbit.c
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/src/signbit.c]:
```suggestion
* int8_t out = stdlib_base_float64_signbit( 3.14 );
```

**Review by kgryte — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/number/float64/base/signbit/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/base/signbit/README.md]:
```suggestion
int8_t out = stdlib_base_float64_signbit( 3.14 );
```

**Review by kgryte — APPROVED** [review]
Thanks, @Pranavchiku!

**Pull request merged**
